joining/src/main/scala/akka/cluster/bootstrap/dns/HeadlessServiceDnsBootstrap.scala

```diff
@@ -0,0 +1,278 @@
+      val baseUri = Uri("http", Uri.Authority(Uri.Host(contactPoint), targetPort))
+      ensureProbing(baseUri)
+    }
+  }
+
+  private def onNoSeedNodesObtainedWithinStableDeadline(contactPoint: Uri): Unit = {
+    val dnsRecordsAreStable = lastContactsObservation.isPastStableMargin(settings, timeNow())
+    if (dnsRecordsAreStable) {
+      lastContactsObservation.selfAddressIfAbleToJoinItself(context.system) match {
+        case Some(allowedToJoinSelfAddress) ⇒
+          log.info(
+              "Initiating new cluster, self-joining [{}], as this node has the LOWEST address out of: [{}]! " +
+              "Other nodes are expected to locate this cluster via continued contact-point probing.")
+
+          cluster.join(allowedToJoinSelfAddress)
+
+          context.stop(self) // the bootstraping is complete
+        case None ⇒
+          log.info(
+              "Exceeded stable margins without locating seed-nodes, however this node is NOT the lowest address out " +
+              "of the discovered IPs in this deployment, thus NOT joining self. Expecting node ({}) to perform the self-join " +
+              "and initiate the cluster.", lastContactsObservation.lowestAddressContactPoint)
+
+        // nothing to do anymore, the probing will continue until the lowest addressed node decides to join itself.
+        // note, that due to DNS changes this may still become this node! We'll then await until the dns stableMargin
+        // is exceeded and would decide to try joining self again (same code-path), that time successfully though.
+      }
+
+    } else {
+      // TODO throttle this logging? It may be caused by any of the probing actors
+      log.info(
+          "DNS observation has changed more recently than the dnsStableMargin({}) allows (at: {}), not considering to join myself. " +
+          "This process will be retried.", settings.dnsStableMargin, new Date(lastContactsObservation.observedAt))
+    }
+  }
+
+  private def ensureProbing(baseUri: Uri): ActorRef = {
+    val childActorName = s"contactPointProbe-${baseUri.authority.host}-${baseUri.authority.port}"
+    log.info("Ensuring probing actor: " + childActorName)
+    context.child(childActorName) match {
+      case Some(contactPointProbingChild) ⇒
+        contactPointProbingChild
+      case None ⇒
+        val props = HttpContactPointBootstrap.props(settings, self, baseUri)
+        context.actorOf(props, childActorName)
+    }
+  }
+
+  private def scheduleNextResolve(serviceName: String, resolveTimeout: FiniteDuration): Unit =
+    timers.startSingleTimer(TimerKeyResolveDNS, Internal.AttemptResolve(serviceName, resolveTimeout),
+      settings.dnsResolveInterval)
+
+  private def resolve(name: String, resolveTimeout: FiniteDuration): Future[DnsResolvedTo] = {
+    def cleanIpString(ipString: String) =
+      ipString.replaceAll("/", "")
+
+    dns.ask(Dns.Resolve(name))(resolveTimeout) map {
+      case srv: SrvResolved =>
```

Comment:
You can extract the ports from `srv`, i.e. `srv.srv.map(v => cleanIpString(v.target) -> v.port)` gives you address and port pair.

Reply:
Related problem, at least in minikube is that I'm getting back A records actually, so I have to figure out why that is, see:

```
[akka.tcp://DemoApp@172.17.0.8:2552/system/IO-DNS/async-dns/$a/$a] Decoded: Message(2,<AN,QUERY,RD,RA,SUCCESS>,Vector(Question(appka-service.default.svc.cluster.local,A,IN)),Vector(ARecord(appka-service.default.svc.cluster.local,30,/172.17.0.6), ARecord(appka-service.default.svc.cluster.local,30,/172.17.0.9), ARecord(appka-service.default.svc.cluster.local,30,/172.17.0.8), ARecord(appka-service.default.svc.cluster.local,30,/172.17.0.7)),Vector(),Vector())
```

Reply:
Perhaps we can try `akka.io.dns.resolve-srv=true` setting? This will enable SRV record resolve, and this setting is set to `false` by default:

https://github.com/ilya-epifanov/akka-dns#advanced-configuration

I'm asking because I can't seem to find this setting being set in the branch.

Reply:
lol, great catch -- I'm blind, missed that setting :)

Reply:
😄 all good 👍